shell/platform/android/io/flutter/embedding/engine/dart/DartMessenger.java

```diff
@@ -86,6 +86,8 @@ public void handleMessageFromDart(
       } catch (Exception ex) {
         Log.e(TAG, "Uncaught exception in binary message listener", ex);
         flutterJNI.invokePlatformMessageEmptyResponseCallback(replyId);
+      } catch (Error err) {
```

Comment:
Should this catch all Throwables?

Reply:
For now, no. I wanted to be explicit about handling `Error`s which is a special kind of throwable that is not supposed to be caught. Java does not (currently) define any `Throwable` other than `Error` and `Exception`.